Provide a share button for users to share document

ppeeou opened the pull request:
<!--  Thanks for sending a pull request! -->

#### What this PR does / why we need it?

Provide a share button for users to share document

![image](https://user-images.githubusercontent.com/10924072/111894264-cdcf6200-8a4c-11eb-88dc-1f1aa2e2317b.png)


#### Any background context you want to provide?


#### What are the relevant tickets?
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #66

### Checklist
- [x] Added relevant tests or not required
- [x] Didn't break anything

hackerwins reviewed (approved):
LGTM. Leave some minor comments.

1. Is there any reason you deleted the package-lock.json?
2. How about renaming `Commons` folder as `commons`?
3. How about renaming `ShareSetting` to `ShareButton`?

ppeeou commented:
> LGTM. Leave some minor comments.
> 
> 1. Is there any reason you deleted the package-lock.json?
> 2. How about renaming `Commons` folder as `commons`?
> 3. How about renaming `ShareSetting` to `ShareButton`?

@hackerwins 
I modified the contents of 1,2 it seems a little better👍

Thanks for the comment.
package-lock.json This was not deleted, but there was a big change..! I'm using npm 6 version before, so it seems to have changed a lot.
(Locked version is different)
https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json#lockfileversion
After changing the npm 7 version, I updated it again, and it seems to have come out as desired! 🙇
